Tasks/Common/nuget-task-common/pe-parser/index.ts

```diff
@@ -13,25 +13,36 @@ export async function getFileVersionInfoAsync(file: string | IReadableFile): Pro
     const readableFile: IReadableFile = typeof file === "string"
         ? await ReadableFile.openAsync(file)
         : file;
-
-    const peFile = await PEImageFile.readAsync(readableFile);
-
-    const resourceSectionTableEntry = peFile.getSection(".rsrc\0\0\0");
-    if (!resourceSectionTableEntry) {
-        throw new PEParserError("noResourceSection", "No resource section found in the file");
-    }
-
-    const resourceSection = await ResourceSection.load(readableFile, resourceSectionTableEntry);
-
-    const versionResourceBuffer = await resourceSection.getResourceBufferAsync(
-        versionResourceTypeId,
-        versionResourceName,
-        neutralLanguageId);
-
-    if (!versionResourceBuffer) {
-        throw new PEParserError("noVersionResource", "No neutral-language version resource found in the file");
+    try {
+        const peFile = await PEImageFile.readAsync(readableFile);
+
+        const resourceSectionTableEntry = peFile.getSection(".rsrc\0\0\0");
+        if (!resourceSectionTableEntry) {
+            throw new PEParserError("noResourceSection", "No resource section found in the file");
+        }
+
+        const resourceSection = await ResourceSection.load(readableFile, resourceSectionTableEntry);
+
+        const versionResourceBuffer = await resourceSection.getResourceBufferAsync(
+            versionResourceTypeId,
+            versionResourceName,
+            neutralLanguageId);
+
+        if (!versionResourceBuffer) {
+            throw new PEParserError("noVersionResource", "No neutral-language version resource found in the file");
+        }
+
+        let versionResource = new VersionResource(versionResourceBuffer);
+        return versionResource.versionInfo;
+    } finally {
+        if (typeof file === "string") {
+            // we opened the file, so we're responsible for closing it
+            await readableFile.closeAsync();
```

Comment:
Add guard for readable file not being null? 
if (typeof file === "string" && readableFile) {....}